### ISSUE TITLE
[Q] how to pass eval_loss from lora_config to wandb.

### ISSUE THREAD

**Issue by LAB-703** [wandb/wandb#5818]
Hi, glad to use wandb.

I'm curious of how to pass eval_loss from lora_config to wandb.
I'm using wandb within https://github.com/tloen/alpaca-lora/blob/main/finetune.py

train_loss and eval_loss can be tracked in the log.
But, I can only pass train_loss but eval_loss to wandb.
How can I pass eval_loss from lora_config to wandb?

Below are the log for your information.

{'loss': 3.2824, 'learning_rate': 1.7999999999999997e-05, 'epoch': 0.04}
 8 {'loss': 3.2785, 'learning_rate': 4.7999999999999994e-05, 'epoch': 0.07}
 9 {'loss': 3.2624, 'learning_rate': 7.5e-05, 'epoch': 0.11}
10 {'loss': 3.0985, 'learning_rate': 0.00010499999999999999, 'epoch': 0.15}
11 {'loss': 2.4684, 'learning_rate': 0.000135, 'epoch': 0.18}
12 {'loss': 1.7627, 'learning_rate': 0.000165, 'epoch': 0.22}
13 {'loss': 1.3802, 'learning_rate': 0.000195, 'epoch': 0.25}
14 {'loss': 1.1985, 'learning_rate': 0.000225, 'epoch': 0.29}
15 {'loss': 1.1125, 'learning_rate': 0.00025499999999999996, 'epoch': 0.33}
16 {'loss': 1.051, 'learning_rate': 0.000285, 'epoch': 0.36}
17 {'loss': 1.022, 'learning_rate': 0.0002997222222222222, 'epoch': 0.4}
18 {'loss': 0.991, 'learning_rate': 0.00029916666666666666, 'epoch': 0.44}
19 {'loss': 0.9743, 'learning_rate': 0.0002986111111111111, 'epoch': 0.47}
20 {'loss': 0.9605, 'learning_rate': 0.00029805555555555553, 'epoch': 0.51}
21 {'loss': 0.9471, 'learning_rate': 0.00029749999999999997, 'epoch': 0.54}
22 {'loss': 0.9361, 'learning_rate': 0.0002969444444444444, 'epoch': 0.58}
23 {'loss': 0.9205, 'learning_rate': 0.00029638888888888884, 'epoch': 0.62}
24 {'loss': 0.9184, 'learning_rate': 0.00029583333333333333, 'epoch': 0.65}
25 {'loss': 0.9345, 'learning_rate': 0.00029527777777777777, 'epoch': 0.69}
26 {'loss': 0.9124, 'learning_rate': 0.0002947222222222222, 'epoch': 0.73}
27   1%|▋                                                                                            | 4/552 [00:01<04:00,  2.28it/s]
28 {'eval_loss': 0.9400128126144409, 'eval_runtime': 287.3962, 'eval_samples_per_second': 15.341, 'eval_steps_per_second': 1.921, 'epoch': 0.73}

**Comment by rsanandres-wandb** [comment]
Hello!

You can subclass the [WandbCallback](https://docs.wandb.ai/ref/python/integrations/keras/wandbcallback) to log what you wish. Here are some links that can be helpful:
- [HuggingFace Transformers Docs](https://docs.wandb.ai/guides/integrations/huggingface)
- [WandbCallBack Docs](https://docs.wandb.ai/ref/python/integrations/keras/wandbcallback)
- [GitHub WandbLogger Code](https://github.com/huggingface/transformers/blob/fe861e578f50dc9c06de33cd361d2f625017e624/src/transformers/integrations.py#L670)

**Comment by LAB-703** [comment]
Hi, thanks for the documents you attached.
I tried to find some cue to solve the problem but I didn't get it.
I think it's because of the difference between 'val_loss' and 'eval_loss'.
Can I change the log? I've tried to change the argument 'monitor' to 'eval_loss' in class WandbCallback(tf.keras.callbacks.Callback)
But, it doesn't work because I've used pytorch.

**Comment by sydholl** [comment]
We are closing this issue due to inactivity. If you feel that it merits additional review, please provide a reproduction and add a comment explaining why. Thank you.
